router/broker.go

```diff
@@ -134,6 +135,38 @@ func (b *broker) publish(pub *wamp.Session, msg *wamp.Publish) {
 		return
 	}
 
+	details := wamp.Dict{}
+
+	// Check and handle Payload PassThru Mode
+	// @see https://wamp-proto.org/wamp_latest_ietf.html#name-payload-passthru-mode
+	if pptScheme, _ := msg.Options[wamp.OptPPTScheme].(string); pptScheme != "" {
+
+		// Let's check: was ppt feature announced by publisher?
+		if !pub.HasFeature(wamp.RolePublisher, wamp.FeaturePayloadPassthruMode) {
+			// It's protocol violation, so we need to abort connection
+			abortMsg := wamp.Abort{Reason: wamp.ErrProtocolViolation}
+			abortMsg.Details = wamp.Dict{}
+			abortMsg.Details[wamp.OptMessage] = "Peer is trying to use Payload PassThru Mode while it was not announced during HELLO handshake"
+			pub.Peer.Send(&abortMsg)
+			pub.Peer.Close()
+
+			return
+		}
+
+		// Every side supports PPT feature
+		// Let's fill PPT options for callee
+		details[wamp.OptPPTScheme] = pptScheme
+		if val, ok := msg.Options[wamp.OptPPTSerializer]; ok {
+			details[wamp.OptPPTSerializer] = val.(string)
+		}
+		if val, ok := msg.Options[wamp.OptPPTCipher]; ok {
+			details[wamp.OptPPTCipher] = val.(string)
+		}
+		if val, ok := msg.Options[wamp.OptPPTKeyId]; ok {
+			details[wamp.OptPPTKeyId] = val.(string)
+		}
```

Comment:
this can be made a helper:
```suggestion
func pptOptionsToDetails(options map[string]inteface{}, details map[string]string) {
	details[wamp.OptPPTScheme] = options[wamp.OptPPTScheme].(string)
	if val, ok := options[wamp.OptPPTSerializer]; ok {
		details[wamp.OptPPTSerializer] = val.(string)
	}
	if val, ok := options[wamp.OptPPTCipher]; ok {
		details[wamp.OptPPTCipher] = val.(string)
	}
	if val, ok := options[wamp.OptPPTKeyId]; ok {
		details[wamp.OptPPTKeyId] = val.(string)
	}
}
```

and then
```
pptOptionsToDetails(msg.Options, details)
```

Reply:
indeed